source/fem-dem/fluid_dynamics_vans_matrix_free_operators.cc

```diff
@@ -0,0 +1,499 @@
+    integrator.evaluate(EvaluationFlags::values | EvaluationFlags::gradients |
+                        EvaluationFlags::hessians);
+  else
+    integrator.evaluate(EvaluationFlags::values | EvaluationFlags::gradients);
+
+  const unsigned int cell = integrator.get_current_cell_index();
+
+  // To identify whether the problem is transient or steady
+  bool transient =
+    (is_bdf(this->simulation_control->get_assembly_method())) ? true : false;
+
+  // Vector for BDF coefficients
+  const Vector<double> *bdf_coefs;
+  if (transient)
+    bdf_coefs = &this->simulation_control->get_bdf_coefficients();
+
+  for (const auto q : integrator.quadrature_point_indices())
+    {
+      Tensor<1, dim, VectorizedArray<number>> source_value;
+
+      // Evaluate source term function if enabled
+      if (this->forcing_function)
+        source_value = this->forcing_terms(cell, q);
+
+      // Add to source term the dynamic flow control force (zero if not
+      // enabled)
+      source_value += this->beta_force;
+
+      // Gather the original value/gradient
+      typename FECellIntegrator::value_type    value = integrator.get_value(q);
+      typename FECellIntegrator::gradient_type gradient =
+        integrator.get_gradient(q);
+      typename FECellIntegrator::gradient_type hessian_diagonal;
+
+      if (this->enable_hessians_jacobian)
+        hessian_diagonal = integrator.get_hessian_diagonal(q);
+
+      // Result value/gradient we will use
+      typename FECellIntegrator::value_type    value_result;
+      typename FECellIntegrator::gradient_type gradient_result;
+      typename FECellIntegrator::hessian_type  hessian_result;
+
+      // Gather previous values of the velocity and the pressure
+      auto previous_values   = this->nonlinear_previous_values(cell, q);
+      auto previous_gradient = this->nonlinear_previous_gradient(cell, q);
+      auto previous_hessian_diagonal =
+        this->nonlinear_previous_hessian_diagonal(cell, q);
+
+      // Gather void fraction value and gradient
+      auto vf_value    = this->void_fraction(cell, q);
+      auto vf_gradient = this->void_fraction_gradient(cell, q);
+
+      Tensor<1, dim + 1, VectorizedArray<number>> previous_time_derivatives;
+      if (transient)
+        previous_time_derivatives =
+          this->time_derivatives_previous_solutions(cell, q);
+
+      // Get stabilization parameter
+      const auto tau = this->stabilization_parameter[cell][q];
+
+      // Weak form Jacobian
+      value_result[dim] = 0;
+      for (unsigned int i = 0; i < dim; ++i)
+        {
+          // ν(∇v,ɛ∇δu)
+          gradient_result[i] =
+            this->kinematic_viscosity * vf_value * gradient[i];
+          // ν(v,∇ɛ∇δu)
+          value_result[i] =
+            this->kinematic_viscosity * vf_gradient * gradient[i];
+          // -(∇·v,ɛδp)
+          gradient_result[i][i] += -vf_value * value[dim];
+          // -(v,δp∇ɛ)
+          value_result[i] += -vf_gradient[i] * value[dim];
```

Comment:
Where is this term coming from? I do not see it in the description above...

Reply:
Yeah this terms was not written above. I added it.